assets/js/dist/dark.js

```diff
@@ -0,0 +1,22 @@
+window.tailwindUIBackend = {
+    isDarkModeEnabled: function() {
+        return (
+            typeof localStorage.tailwindUIBackendDarkMode !== 'undefined'
+            ? JSON.parse(localStorage.tailwindUIBackendDarkMode)
+            : window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
+        );
+    },
+    toggleDarkMode: function() {
+        localStorage.tailwindUIBackendDarkMode = !window.tailwindUIBackend.isDarkModeEnabled();
+        window.tailwindUIBackend.updateDarkMode();
+    },
+    updateDarkMode: function () {
+        if (window.tailwindUIBackend.isDarkModeEnabled()) {
+            document.documentElement.classList.add('dark');
+        } else {
+            document.documentElement.classList.remove('dark');
+        }
+    },
+};
+
+tailwindUIBackend.updateDarkMode();
```

Comment:
Instead of storing this in the localStorage, can we store / load it from UserPreferences (the same place that timezone and locale are set)?

Reply:
I was first thinking about that but this would mean to create a controller + ajax handler for it and I wanted a simple solution first? Or is there already an easy way of doing that?

Reply:
Off the top of my head you could probably do something like the following:
```php
\Backend\Classes\Controller::extend(function ($controller) {
    $controller->addDynamicMethod('onChangeDarkMode', function () {
        \Backend\Models\UserPreference::set('dark_mode', post('mode'));
    });
});
```

And then just add relevant fields to the backend user preferences form as well as load the setting from the userpreference model

Reply:
I've now refactured it to use user preferences, added the dark_mode field to the preference model form and the ajax handler. I've however removed the fallback to prefered color sheme, as it would have made things overly complex and would have required more JS.

Reply:
By preferred color scheme you mean whatever the user's requested in their OS settings? Ideally I'd like to retain that functionality, have the value stored as `auto` and have it be the default behaviour if nothing else has been set.